static/js/demos.js

```diff
@@ -92,6 +92,19 @@ function update(computer, instructions, num_code_lines, highlighted_registers) {
     $('#line_desc')[0].innerHTML = line_desc;
 }
 
+function populate_code(instructions) {
```

Comment:
`const i` in the for-in declaration (new context per run through the loop, so we're not actually reassigning the variable), please...

Can you try rewriting the loop using the template literal syntax? It's the same kind of thing as f-strings in Python. It usually makes this kind of programmatic generation of HTML a lot more readable. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

(Sometimes it ends up messier, but it's worth knowing about...)